query-engine/connector-test-kit/src/test/scala/queries/batch/TransactionalBatchSpec.scala

```diff
@@ -82,7 +82,7 @@ class TransactionalBatchSpec extends FlatSpec with Matchers with ApiSpecBase {
     )
 
     server.batch(queries, transaction = false, project, legacy = false).toString should startWith(
-      """[{"errors":[{"error":"Error occurred during query execution:\nConnectorError(ConnectorError { user_facing_error: Some(KnownError { message: \"Unique constraint failed"""
+      """{"batchResult":[{"errors":[{"error":"Error occurred during query execution:\nConnectorError(ConnectorError { user_facing_error: Some(KnownError { message: \"Unique constraint failed"""
```

Comment:
I'm not sure exactly the difference between the top-level and the nested ones. I assume that the:
- top-level `errors` field is for stuff that happens when it's not even possible to run the query
- the nested `errors` one is for when a query could be run but something unexpected happened after that.

According to that logic, the result seems right here but still feels clunky to me, that we sometimes have errors buried in the `batchResult` and sometimes have them top-level.

Reply:
Should we maybe always have all the errors gathered on the top-level `errors` field and remove the nested `errors` field that lives inside the `GQLResponse` type? Leaving us with such a shape:

```rust
struct GQLBatchResponse {
  batch_result: Vec<Map>,
  errors: Vec<GQLError>
}
```

Reply:
Your assumption is correct, but this will not be changed. Leave it as-is.